src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlDataReader.cs

```diff
@@ -5824,18 +5640,8 @@ private static Task<T> GetFieldValueAsyncExecute<T>(Task task, object state)
             {
                 if (reader.IsCommandBehavior(CommandBehavior.SequentialAccess) && reader._sharedState._dataReady)
                 {
-                    bool internalReadSuccess = false;
-                    TdsParser.ReliabilitySection tdsReliabilitySection = new TdsParser.ReliabilitySection();
                     RuntimeHelpers.PrepareConstrainedRegions();
```

Comment:
This is only useful if it's immediately followed by a try block (see the Remarks section of the [PrepareConstrainedRegions](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.runtimehelpers.prepareconstrainedregions?view=netframework-4.6.2#remarks) documentation). It can be removed.

Reply:
Good catch 👍